src/gt4py/backend/gtc_backend/common.py

```diff
@@ -57,21 +57,24 @@ def pybuffer_to_sid(
     domain_dim_flags: Tuple[bool, bool, bool],
     data_ndim: int,
     stride_kind_index: int,
+    check_layout: bool = True,
     backend,
 ):
```

Comment:
It's a little nit of mine, but I prefer arguments (even inside the kwargs list) with defaults to come after those without. So, switch `backend` and `check_layout`. It's a bit orthogonal to the PR, but could you add a type hint to `backend` here as well as `_get_unit_stride_dim` above? Thanks!